lib/UI/calendar/widgets/logout_button.dart

```diff
@@ -19,28 +20,59 @@ class _LogoutButtonState extends State<LogoutButton> {
     return Padding(
       padding: const EdgeInsets.only(right: 16.0),
       child: InkWell(
-        onTap: () async {
-          if (!_isLoading) {
-            setState(() {
-              _isLoading = true;
-            });
-            await FirebaseAuth.instance.signOut();
-            setState(() {
-              _isLoading = false;
-            });
-            Navigator.pushAndRemoveUntil(
-              context,
-              MaterialPageRoute(builder: (context) => MyApp()),
-              (route) => false,
-            );
-          }
+        onTap: () {
+          _checkShouldSignOut();
         },
         child: _isLoading
-            ? Center(
-                child: CircularProgressIndicator(strokeWidth: 2),
-              )
+            ? Center(child: CircularProgressIndicator(strokeWidth: 2))
             : Icon(Icons.logout),
       ),
     );
   }
+
+  void _navigateToLandingScreen(BuildContext context) {
+    Navigator.pushAndRemoveUntil(
+      context,
+      MaterialPageRoute(builder: (context) => LandingScreen()),
+      (route) => false,
+    );
+  }
+
+  void _checkShouldSignOut() async {
+    if (!_isLoading) {
+      setState(() => _isLoading = true);
+      bool _shouldLogOut = false;
+      await getDialogValue(
+        title: 'Sign Out',
+        description:
+            'You have attempted to sign out. Are you sure you want to sign out of ExerLog?',
+        actions: [
+          TextButton(
+            onPressed: () {
+              _shouldLogOut = false;
+              _navigatorPop();
+            },
+            child: Text('Cancel'),
+          ),
+          TextButton(
+            onPressed: () {
+              _shouldLogOut = true;
+              Navigator.pop(context);
+            },
+            child: Text('Sign Out'),
+          ),
+        ],
+        context: context,
```

Comment:
Consider making a mixin were we store all the showing alert methods (only 1 now, this one). And make this mixin on the state of stateful widgets. This way you don't have to pass the context each time.

Reply:
I don't think I understand what "making a mixin on the state of stateful widgets" means. Right now, I have a mixin called `Dialogs`

Path: lib > mixins > dialogs.dart

```dart
import 'package:exerlog/src/widgets/theme/theme_provider.dart';
import 'package:flutter/material.dart';

mixin Dialogs {
  Widget _dialog({
    required String title,
    required String description,
    required List<Widget> actions,
  }) {
    return ThemeProvider(
      builder: (context, theme) => new AlertDialog(
        backgroundColor: theme.colorTheme.backgroundColorVariation,
        title: Text(title),
        content: Text(description),
        actions: actions,
      ),
    );
  }

  Future<bool> showSignoutConfirmationDialog() async {
    // Will use _dialog to show dialog. Need BuildContext access here.
  }
}
```

Reply:
I don't think we've discussed yet how we're gonna handle dialogs, so this is just my opinion. but you can make a mixin on a stateful widget like this. 

```
mixin AlertMixin<T extends StatefulWidget> on State<T> {
  void showConfirmationDialog() => // show dialog here, you can use context because it's automatically gonna use the context of the state you use this mixin on;
}
```
Then use it on a stateful widget like
```
class _ExampleState extends State<Example> with AlertMixin {

  future<void> _onLogoutPressed()async{
   final result = await showConfirmationDialog()
  // handle the rest
 } 
} 
```